packages/autocomplete-core/src/utils/isAndroid.ts

```diff
@@ -0,0 +1,11 @@
+const androidRegex = /android[-/ ]?([\w.]*)/i;
+const windowsPhoneRegex = /(windows (?:phone(?: os)?|mobile))[/ ]?([\d.\w ]*)/i;
+
+export function isAndroid(userAgent: string) {
+  return (
+    userAgent &&
+    Boolean(
+      userAgent.match(androidRegex) && !userAgent.match(windowsPhoneRegex)
+    )
+  );
+}
```

Comment:
This ensures the functions always returns a boolean.

```suggestion
export function isAndroid(userAgent: string) {
  return Boolean(
    userAgent &&
      userAgent.match(androidRegex) &&
      !userAgent.match(windowsPhoneRegex)
  );
}
```